lti_consumer/tests/unit/plugin/test_views.py

```diff
@@ -687,12 +721,65 @@ def test_access_token_endpoint_with_location_in_url(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.json(), token)
 
-    def test_non_existant_configuration_for_given_id(self):
+    @patch('lti_consumer.plugin.views.get_lti_api_base')
+    @patch('lti_consumer.plugin.views.LtiConsumer1p3')
+    @patch('lti_consumer.plugin.views.get_external_config_from_filter')
+    def test_access_token_endpoint_with_external_id_in_url(
+        self,
+        get_external_config_from_filter,
+        lti_consumer,
+        get_lti_api_base,
+    ):
+        """
+        Check that the access_token generated by the lti_consumer is returned.
+        """
+        external_config = {
+            'version': LtiConfiguration.LTI_1P3,
+            'lti_1p3_client_id': 'client-id',
+            'lti_1p3_private_key': 'private-key',
+            'lti_1p3_private_key_id': 'private-key-id',
+            'lti_1p3_tool_public_key': 'public-key',
+            'lti_1p3_tool_keyset_url': 'keyset-url',
+        }
+        token = {'access_token': 'test-token'}
+        get_external_config_from_filter.return_value = external_config
+        lti_consumer.return_value.access_token.return_value = token
+
+        body = self.get_body(create_jwt(self.key, {}))
+        response = self.client.post(
+            reverse('lti_consumer:lti_consumer.access_token_via_external_id', args=['x', 'x']),
+            data=body,
+        )
+
+        get_external_config_from_filter.assert_called_once_with({}, 'x:x')
+        get_lti_api_base.assert_called_once_with()
+        lti_consumer.assert_called_once_with(
+            iss=get_lti_api_base(),
+            lti_oidc_url=None,
+            lti_launch_url=None,
+            client_id=external_config['lti_1p3_client_id'],
+            deployment_id=None,
+            rsa_key=external_config['lti_1p3_private_key'],
+            rsa_key_id=external_config['lti_1p3_private_key_id'],
+            redirect_uris=None,
+            tool_key=external_config['lti_1p3_tool_public_key'],
+            tool_keyset_url=external_config['lti_1p3_tool_keyset_url'],
+        )
+        lti_consumer().access_token.called_once_with(body)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), token)
+
+    @ddt.data(
+        ('lti_consumer:lti_consumer.access_token', ['075194d3-6885-417e-a8a8-6c931e272f00']),
+        ('lti_consumer:lti_consumer.access_token_via_location', ['block-v1:x+x+x+type@problem+block@x']),
+        ('lti_consumer:lti_consumer.access_token_via_external_id', ['x', 'x']),
+    )
+    @ddt.unpack
+    def test_non_existant_configuration(self, url, args):
```

Comment:
Typo: `nonexistent`. It seems to be occurring in more places in this file.

Reply:
I fixed all occurrences of this typo on the tests in this file on the latest commit.